apps/studio/src/server/modules/auth/singpass/singpass.service.ts

```diff
@@ -0,0 +1,83 @@
+import type { z } from "zod"
+import { generators, Issuer } from "openid-client"
+
+import type { singpassStateSchema } from "~/schemas/auth/singpass"
+import { env } from "~/env.mjs"
+import {
+  SINGPASS_ENCRYPTION_JWK,
+  SINGPASS_REDIRECT_URI,
+  SINGPASS_SCOPES,
+  SINGPASS_SIGN_IN_STATE,
+  SINGPASS_SIGNING_JWK,
+} from "./singpass.constants"
+import { extractUuid } from "./singpass.utils"
+
+const singpassIssuer = await Issuer.discover(env.SINGPASS_ISSUER_ENDPOINT)
+const singpassClient = new singpassIssuer.Client(
+  {
+    client_id: env.SINGPASS_CLIENT_ID,
+    response_types: ["code"],
+    token_endpoint_auth_method: "private_key_jwt",
+    id_token_signed_response_alg: "ES256",
+  },
+  {
+    keys: [SINGPASS_SIGNING_JWK, SINGPASS_ENCRYPTION_JWK],
+  },
+)
+
+export const getAuthorizationUrl = () => {
```

Comment:
nit: might be misleading since this returns an object containing both `authorizationUrl` and `session` instead of just `authorizationUrl`